tests/ui/suspicious_operation_groupings.stderr

```diff
@@ -6,11 +6,23 @@ LL |         self.x == other.y && self.y == other.y && self.z == other.z
    |
    = note: `-D clippy::suspicious-operation-groupings` implied by `-D warnings`
 
+error: This sequence of operators looks suspiciously like a bug.
+  --> $DIR/suspicious_operation_groupings.rs:11:9
+   |
+LL |         self.x == other.y && self.y == other.y && self.z == other.z
```

Comment:
This lint message is emitted twice, and my best guess as to why, is that both `self.x == other.y && self.y == other.y && self.z == other.z` and `self.x == other.y && self.y == other.y` have detectable errors, and they both get visited separately by the lint pass. 

I'm unaware of a good way to prevent these double messages. Is this something that has been addressed for other lints before?

Reply:
I can't remember now a similar case. I think you can leave it like this for now and once we get to review we can try to find a solution.

Reply:
How do we prevent the sub-expressions from triggering the lint again after the parent expression collects all the suggestions after that? As far as I can tell `check_expr` gets called for each `expr` and each `expr` contained inside that `expr`.